warehouse/models/mart/transit_database/dim_service_components.sql

```diff
@@ -0,0 +1,68 @@
+{{ config(materialized='table') }}
+
+WITH latest AS (
+    SELECT *
+    FROM {{ ref('int_transit_database__service_components_unnested') }}
+),
+
+dim_components AS (
+    SELECT * FROM {{ ref('dim_components') }}
+),
+
+dim_services AS (
+    SELECT * FROM {{ ref('dim_services') }}
+),
+
+dim_products AS (
+    SELECT * FROM {{ ref('dim_products') }}
+),
+
+ranked_service_components AS (
+    SELECT
+        key,
+        service_key,
+        product_key,
+        component_key,
+        ntd_certified,
+        product_component_valid,
+        notes,
+        ROW_NUMBER() OVER
+            (PARTITION BY service_key, product_key, component_key
+            ORDER BY key) AS rank,
+        calitp_extracted_at
+    FROM latest
+),
+
+unique_service_components AS (
+    SELECT
+        * EXCEPT(rank)
+    FROM ranked_service_components
+    WHERE rank = 1
+),
```

Comment:
nice, apparently bigquery does support QUALIFY now, could use that here

https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#qualify_clause

Reply:
awesome, updated: https://github.com/cal-itp/data-infra/pull/1625/commits/411d021ffabac2eeec92992bc0785b5908c6a8c3